apps/api/src/server/applications/key-dates/key-dates.controller.js

```diff
@@ -38,9 +39,25 @@ export const updateKeyDates = async ({ body, params }, response) => {
 
 	const keyDates = mapRequestToKeyDates(body);
 
+	const originalCase = await caseRepository.getById(id, {
```

Comment:
This looks similar to `application.controller.js` - are there some service or repo methods we should be adding to hold this logic?

Reply:
I've made the change below and done it in a common util so there's no duplication.